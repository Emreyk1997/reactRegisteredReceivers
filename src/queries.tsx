// import gql from 'graphql-tag';
// import { useQuery } from '@apollo/react-hooks';

// export const query = gql`
//   query Books {
//     books {
//       title
//       author
//     }
//   }
// `;

// export default () => useQuery(query);

import gql from "graphql-tag";
import { useQuery } from '@apollo/react-hooks';

export const GET_COUNTER = gql`
  query GetCounterValue {
    counter @client
  }
`;

export default () => useQuery(GET_COUNTER);
