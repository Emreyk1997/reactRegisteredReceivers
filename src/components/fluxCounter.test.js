import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CounterFlux from './fluxCounter';
import { MockedProvider } from '@apollo/react-testing';
import { GET_COUNTER, GET_CONFIGS } from '../queries';
import { act } from 'react-dom/test-utils';

const CounterMocks = [
  {
    request: {
      query: GET_COUNTER,
    },
    result: {
      data: {
        counter: 0,
      },
    },
  },
];
const ConfigMocks = [
  {
    request: {
      query: GET_CONFIGS,
    },
    result: {
      data: {
        configs: 'hello',
      },
    },
  },
];

describe('Take a snapshot', () => {
  afterEach(cleanup);

  it('should take a snapshot', async () => {
    const promise = Promise.resolve();
    const { asFragment } = render(
      <MockedProvider mocks={CounterMocks} addTypename={false}>
        <CounterFlux />
      </MockedProvider>
    );

    // expect(
    //   asFragment(
    //     <MockedProvider mocks={CounterMocks} addTypename={false}>
    //       <CounterFlux />
    //     </MockedProvider>
    //   )
    // ).toMatchSnapshot();
    await act(() => promise);
  });
});
