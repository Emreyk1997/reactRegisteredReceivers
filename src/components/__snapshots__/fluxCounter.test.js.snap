// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Provider:
    </p>
  </div>
</DocumentFragment>
`;
